src/devtools/client/webconsole/components/Input/JSTerm.tsx

```diff
@@ -86,10 +86,18 @@ export default function JSTerm({
     } else if (e.key === Keys.ARROW_UP && editor.getCursor().line === 0) {
       e.preventDefault();
 
-      if (historyIndex !== null && historyIndex > 0) {
-        const newIndex = historyIndex - 1;
-        const newValue = terminalExpressionHistory[newIndex];
+      let newIndex = null;
+      let newValue = null;
+
+      if (historyIndex === null) {
+        newIndex = terminalExpressionHistory.length - 1;
+        newValue = terminalExpressionHistory[newIndex];
+      } else {
+        newIndex = historyIndex - 1;
+        newValue = terminalExpressionHistory[newIndex];
+      }
 
```

Comment:
This was the gist of the regression. I wasn't properly handling `null` state (the default/initial value). Ooof.